torch/_torch_docs.py

```diff
@@ -4709,6 +4709,45 @@ def merge_dicts(*dicts):
     tensor([  2.,   4.,   8.,  16.])
 """.format(**common_args))
 
+add_docstr(torch.nanprod,
+           r"""
+nanprod(input, dtype=None) -> Tensor
+
+Returns the product of all elements excluding ``NaN`` values in the
+:attr:`input` tensor.
+
+Args:
+    {input}
+    {dtype}
+
+Example::
+
+    >>> a = torch.tensor([float('nan'), 2, 3, 6, float('nan')])
+    >>> out = torch.nanprod(a)
+    >>> out
+    tensor(36.)
+
+.. function:: nanprod(input, dim, keepdim=False, dtype=None) -> Tensor
+
+Returns the product of each row of the :attr:`input` tensor in the given
+dimension :attr:`dim` excluding `NaN` values.
+
+{keepdim_details}
+
+Args:
+    {input}
+    {dim}
+    {keepdim}
+    {dtype}
```

Comment:
Will this autogenerate the same message as the one appearing for torch.prod?

"the desired data type of returned tensor. If specified, the input tensor is casted to dtype before the operation is performed. This is useful for preventing data type overflows. Default: None."